storybookjs/storybook
# feat: change the file structure in templates used in `storybook init`

Closes #22151



## What I did

I updated the file structure for the CLI templates used in `storybook init`. My goal was to have the story files and component files organized together. The proposed file structure for all templates is as follows:

```tree
├── stories
│   ├── introduction.mdx
│   ├── Button
│   │   ├── index.jsx
│   │   ├── Button.stories.js
│   │   ├── button.css
│   ├── Header
│   │   ├── index.jsx
│   │   ├── Header.stories.js
│   │   ├── header.css
│   ├── Page
│   │   ├── index.jsx
│   │   ├── Page.stories.js
│   │   ├── page.css
```

## How to test

1. Run a sandbox for the template, e.g. `yarn task --task sandbox --start-from auto --template react-vite/default-ts`
2. Check the file structure in the stories folder.
3. Open Storybook in your browser.
4. Access to any stories.

## Checklist



      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

#### Maintainers

      make sure to add the `ci:merged` or `ci:daily` GH label to it.

`["cleanup", "BREAKING CHANGE", "feature request", "bug", "documentation", "maintenance", "dependencies", "other"]`
